sympy/ntheory/primetest.py

```diff
@@ -4,42 +4,41 @@
 """
 
 from __future__ import print_function, division
-from sympy.core.compatibility import range
-
-# pseudoprimes that will pass through last mr_safe test
-_pseudos = set([
-            669094855201,
-           1052516956501, 2007193456621, 2744715551581, 9542968210729,
-          17699592963781, 19671510288601,
-          24983920772821, 24984938689453, 29661584268781, 37473222618541,
-          46856248255981, 47922612926653, 48103703944453, 49110566041153,
-          49752242681221, 91206655032481, 91481980096033, 119034193492321,
-         123645258399601, 128928036060253, 137364148720147, 150753857310253,
-         153131886327421, 155216912613121, 185610214763821, 224334357392701,
-         227752294950181, 230058334559041, 304562854940401, 306001576998253,
-         335788261073821, 377133492079081, 379242177424951, 389970770948461,
-         397319638319521, 448114903362253, 523235160050221, 628999496281621,
-         699349238838253, 746667678235753, 790198268451301, 794036495175661,
-         823820871230281, 867739535711821, 1039918661294761, 1099127938585141,
-        1104388025338153, 1173374598605653, 1262797719066157, 1265872947674653,
-        1325898212229667, 1327034517143653, 1418575746675583, 1666122072463621,
-        1837400535259453, 1857422490084961, 1870756820971741, 1914550540480717,
-        2018963273468221, 2163829000939453, 2206020317369221, 2301037384029121,
-        2416062055125421, 2435076500074921, 2545656135020833, 2594428516569781,
-        2669983768115821, 2690937050990653, 2758640869506607, 2833525461416653,
-        2876662942007221, 2932155806957821, 2957010595723801, 3183606449929153,
-        3220133449185901, 3424103775720253, 3625360152399541, 3939300299037421,
-        3947917710714841, 3980273496750253, 4182256679324041, 4450605887818261,
-        4727893739521501, 4750350311306953, 4755334362931153, 5756440863559753,
-        5760976603475341, 5794399356078761, 5954850603819253, 6125544931991761,
-        6320931714094861, 6347593619672581, 6406268028524101, 6510632945054941,
-        6620082224794741, 6627325072566061, 6844056606431101, 6989404981060153,
-        7144293947609521, 7288348593229021, 7288539837129253, 7406102904971689,
-        7430233301822341, 7576425305871193, 7601696719033861, 7803926845356487,
-        7892007967006633, 7947797946559453, 8207000460596953, 8295064717807513,
-        8337196000698841, 8352714234009421, 8389755717406381, 8509654470665701,
-        8757647355282841, 8903933671696381, 8996133652295653, 9074421465661261,
-        9157536631454221, 9188353522314541])
+
+from sympy.core.compatibility import range, as_int
+from sympy.core.numbers import Float
+
+from mpmath.libmp import bitcount as _bitlength
+
+
+
+def _int_tuple(*i):
+    return tuple(int(_) for _ in i)
+
+
+def is_square(n, prep=True):
```

Comment:
> Perhaps it would also be worthwhile to add the other filters

Perhaps. I would like to see this committed and let someone else tackle that if there is interest.